src/Payload.cpp

```diff
@@ -45,20 +53,28 @@ void Payload::serializeInto(folly::io::QueueAppender& appender) {
 std::unique_ptr<folly::IOBuf> Payload::deserializeMetadataFrom(
     folly::io::Cursor& cur,
     FrameFlags flags) {
-  std::unique_ptr<folly::IOBuf> metadata;
-
-  if (flags & FrameFlags_METADATA) {
-    const auto length = cur.readBE<uint32_t>();
+  if ((flags & FrameFlags_METADATA) == 0) {
+    return nullptr;
+  }
 
-    assert(length < kMaxMetadataLength);
-    (void)kMaxMetadataLength;
+  const auto length = cur.readBE<uint32_t>();
 
-    const auto metadataPayloadLength = length - sizeof(uint32_t);
+  if (length >= kMaxMetadataLength) {
+    throw std::runtime_error("Metadata is too big to deserialize");
+  }
 
-    if (metadataPayloadLength > 0) {
-      cur.clone(metadata, metadataPayloadLength);
-    }
+  if (length <= sizeof(uint32_t)) {
```

Comment:
Ignore, it has changed in version 1. so I think we are ok until we adopt the new framing format.

https://github.com/ReactiveSocket/reactivesocket/blob/0.2/Protocol.md